packages/wrapped-keys-lit-actions/src/lib/common/batchGenerateEncryptedKeys.js

```diff
@@ -0,0 +1,159 @@
+const { encryptPrivateKey } = require('./internal/encryptKey');
+const {
+  generateEthereumPrivateKey,
+} = require('../ethereum/internal/generatePrivateKey');
+const { signMessageEthereumKey } = require('../ethereum/internal/signMessage');
+const {
+  generateSolanaPrivateKey,
+} = require('../solana/internal/generatePrivateKey');
+const { signMessageSolanaKey } = require('../solana/internal/signMessage');
+
+/* global accessControlConditions, actions, Lit*/
+
+async function processEthereumAction(action, ndx) {
+  const { network, generateKeyParams } = action;
+  const messageToSign = action.signMessageParams?.messageToSign;
+
+  const ethereumKey = generateEthereumPrivateKey();
+
+  const [generatedPrivateKey, messageSignature] = await Promise.all([
+    Lit.Actions.runOnce(
+      { waitForResponse: true, name: `encryptEthereumPrivateKey_${ndx}` },
+      async () =>
+        JSON.stringify(
+          await encryptPrivateKey({
+            accessControlConditions,
+            publicKey: ethereumKey.publicKey,
+            privateKey: ethereumKey.privateKey,
+          })
+        )
+    ),
+    messageToSign
+      ? signMessageEthereumKey({
+          messageToSign: messageToSign,
+          privateKey: ethereumKey.privateKey,
```

Comment:
`ethereumKey` will be different for each node so the `messageSignature` won't match the returned `generatedPrivateKey`

Reply:
:+1: https://github.com/LIT-Protocol/js-sdk/pull/675#discussion_r1792721090